NTupleAnalyzer_nanoAOD.py

```diff
@@ -881,6 +889,204 @@ def PropagatePTChangeToMET(met,original_object,varied_object):
+		# https://twiki.cern.ch/twiki/bin/view/CMS/HighPtMuonReferenceRun2
+		if abseta>=0.00 and abseta<=1.60: recoSFbyPt = [[0.9943, 0.0007],[0.9948, 0.0007],[0.9950, 0.0009],[0.994, 0.001],[0.9914, 0.0009],[0.993, 0.002],[0.991, 0.004],[1.0, 0.1]]
+		elif abseta>1.60 and abseta<=2.40: recoSFbyPt = [[1.00, 0.00],[0.993, 0.001],[0.990, 0.001],[0.988, 0.001],[0.981, 0.002],[0.983, 0.003],[0.978, 0.006],[0.98, 0.03]]
+		# Muon high pt identification scale factors
+		# https://gitlab.cern.ch/cms-muonPOG/MuonReferenceEfficiencies/blob/master/EfficienciesStudies/2018/jsonfiles/RunABCD_SF_ID.json
+		if abseta>=0.00 and abseta<=0.90: highPtIdSFbyPt = [[0.992, 0.003],[0.992, 0.001],[0.9923, 0.0008],[0.9927, 0.0005],[0.9929, 0.0006],[0.992, 0.003]]
+		elif abseta>0.90 and abseta<=1.20: highPtIdSFbyPt = [[0.988, 0.005],[0.986, 0.002],[0.986, 0.003],[0.9869, 0.0005],[0.986, 0.001],[0.985, 0.002]]
+		elif abseta>1.20 and abseta<=2.10: highPtIdSFbyPt = [[0.9922, 0.0009],[0.992, 0.002],[0.992, 0.001],[0.9916, 0.0002],[0.9925, 0.001],[0.991, 0.0007]]
+		elif abseta>2.10 and abseta<=2.40: highPtIdSFbyPt = [[0.98, 0.004],[0.978, 0.003],[0.978, 0.001],[0.978, 0.002],[0.977, 0.003],[0.974, 0.005]]
+		# Muon relative track isolation scale factors
+		# https://gitlab.cern.ch/cms-muonPOG/MuonReferenceEfficiencies/blob/master/EfficienciesStudies/2018/jsonfiles/RunABCD_SF_ISO.json
+		if abseta>=0.00 and abseta<=0.90: relTrkIsoSFbyPt = [[0.992, 0.004],[0.991, 0.002],[0.9945, 0.0005],[0.9968, 0.0005],[0.9969, 0.0006],[0.9975, 0.0007]]
+		elif abseta>0.90 and abseta<=1.20: relTrkIsoSFbyPt = [[0.998, 0.006],[0.994, 0.003],[0.9979, 0.0009],[0.9994, 0.0005],[0.9997, 0.0009],[1.0, 0.001]]
+		elif abseta>1.20 and abseta<=2.10: relTrkIsoSFbyPt = [[1.007, 0.003],[1.005, 0.002],[1.0032, 0.0006],[1.0026, 0.0006],[1.0011, 0.0007],[1.0011, 0.0009]]
+		elif abseta>2.10 and abseta<=2.40: relTrkIsoSFbyPt = [[1.018, 0.005],[1.008, 0.003],[1.009, 0.003],[1.005, 0.0009],[1.004, 0.001],[1.002, 0.002]]
+		# High level trigger scale factors
+		# https://twiki.cern.ch/twiki/bin/view/CMS/HighPtMuonReferenceRun2
+		if abseta>=0.00 and abseta<=0.90: hltSFbyPt = [[0.980, 0.006],[0.982, 0.004],[0.976, 0.001],[0.980, 0.002],[0.978, 0.004],[0.973, 0.009],[0.957, 0.03]]
+		elif abseta>0.900 and abseta<=1.20: hltSFbyPt = [[0.982, 0.01],[0.966, 0.005],[0.967, 0.002],[0.964, 0.004],[0.960, 0.009],[0.985, 0.02],[0.988, 0.06]]
+		elif abseta>1.20 and abseta<2.10: hltSFbyPt = [[0.983, 0.01],[1.012, 0.008],[1.002, 0.003],[1.004, 0.004],[1.009, 0.009],[0.993, 0.01],[1.062, 0.04]]
+		elif abseta>2.10 and abseta<2.40: hltSFbyPt = [[0.987, 0.03],[0.952, 0.02],[0.999, 0.008],[1.003, 0.01],[1.024, 0.02],[0.993, 0.05],[1.070, 0.2]]
+
+	# Check muon pt and select SF from appropriate pt bin. Different SFs (e.g., reco vs hlt) have different pt ranges/binning. 
+	# Maps ensure double checking of pt bins do not occur (speeds up loop) for different SF types
+	lowBndMap1 = {0:0, 1:2, 2:3, 3:4, 4:5, 5:5, 6:6, 7:6}
```

Comment:
Please add one more comment:

# This should be checked carefully if any binning changes

Reply:
I take it you mean if we change the scale factors we use (might change pt binning), check the binning?

Reply:
Yes don't change the code, just add a warning comment "# This should....."

Reply:
Ok, will do